camel-core/src/test/java/org/apache/camel/language/simple/SimpleTest.java

```diff
@@ -479,9 +474,26 @@ public void testDateAndTimeExpressions() throws Exception {
         cal.set(Calendar.MILLISECOND, 123);
         exchange.getIn().setHeader("birthday", cal.getTime());
 
+        assertExpression("date:header.birthday - 10s:yyyy-MM-dd'T'HH:mm:ss:SSS", "1974-04-20T08:55:37:123");
         assertExpression("date:header.birthday:yyyy-MM-dd'T'HH:mm:ss:SSS", "1974-04-20T08:55:47:123");
     }
 
+    public void testDateWithTimezone() throws Exception {
+        Calendar cal = Calendar.getInstance();
+        cal.setTimeZone(TimeZone.getTimeZone("GMT+2"));
+        cal.set(1974, Calendar.APRIL, 20, 8, 55, 47);
+        cal.set(Calendar.MILLISECOND, 123);
+        exchange.getIn().setHeader("birthday", cal.getTime());
+
+        assertExpression("date:header.birthday:yyyy-MM-dd'T'HH:mm:ss:SSS", "1974-04-20T08:55:47:123");
+        assertExpression("date-with-timezone:header.birthday:UTC:yyyy-MM-dd'T'HH:mm:ss:SSS", "1974-04-20T06:55:47:123");
+    }
+
+    public void testDatePredicates() throws Exception {
+        assertPredicate("${date:now} < ${date:now+60s}");
+        assertPredicate("${date:now-2s+2s} == ${date:now}");
```

Comment:
LOL, hopefully we keep it to only + and - signs, and not add * or / as well to really confuse users ;)